src/host/screenInfo.cpp

```diff
@@ -40,8 +40,6 @@ SCREEN_INFORMATION::SCREEN_INFORMATION(
     HWheelDelta{ 0 },
     _textBuffer{ nullptr },
     Next{ nullptr },
-    WriteConsoleDbcsLeadByte{ 0, 0 },
-    FillOutDbcsLeadChar{ 0 },

```

Comment:
Sorry, to maintain compatibility, the state really needs to be captured and stored on each individual `SCREEN_INFORMATION`. Not as a whole in the API servicing call in a static.

Reply:
Also please author a test to that effect. Create a 2nd screen buffer and ensure that writing a partial into one of them isn't consumed by writing into the other one and vice versa.

Reply:
Oh, this is somehow surprising. How did it ever work for UTF-8 in the past? I'm already aware that the handling of other codepages is completely inconsistent compared to UTF-8. And I have on my list to investigate why this is. I didn't expect it was that worse though. It won't fit into this PR in total, However, I understand now that I already changed the behavior ...

Reply:
> Also please author a test to that effect.

```
// ScreenBufferTest.c
#include <windows.h>

int main(void)
{
  SetConsoleCP(CP_UTF8);
  SetConsoleOutputCP(CP_UTF8);

  DWORD written = 0;
  HANDLE hStdout = GetStdHandle(STD_OUTPUT_HANDLE);
  HANDLE hNewScreenBuffer = CreateConsoleScreenBuffer(GENERIC_READ | GENERIC_WRITE, FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, CONSOLE_TEXTMODE_BUFFER, NULL);

  WriteConsoleA(hStdout, "\xF0", 1ul, &written, NULL);
  WriteConsoleA(hNewScreenBuffer, "\xF0", 1ul, &written, NULL);
  WriteConsoleA(hNewScreenBuffer, "\x9F", 1ul, &written, NULL);
  WriteConsoleA(hStdout, "\x9F", 1ul, &written, NULL);
  WriteConsoleA(hStdout, "\x98", 1ul, &written, NULL);
  WriteConsoleA(hNewScreenBuffer, "\x91", 1ul, &written, NULL);
  WriteConsoleA(hNewScreenBuffer, "\x8D", 1ul, &written, NULL);
  WriteConsoleA(hStdout, "\x80", 1ul, &written, NULL);

  Sleep(2000ul);
  SetConsoleActiveScreenBuffer(hNewScreenBuffer);
  Sleep(2000ul);
}
```

Output of commit 11602df
![Screenshot (2)](https://user-images.githubusercontent.com/46659645/76147115-47421f80-6099-11ea-9d42-b4e0e303e8ce.png)
... after 2s in the second screen buffer ...
![Screenshot (3)](https://user-images.githubusercontent.com/46659645/76147131-6640b180-6099-11ea-989a-3ee094c693db.png)

I don't have any idea of how to make a unit test out of it.